lib/train/file.rb

```diff
@@ -147,5 +129,71 @@ def mounted?
 
       !mounted.nil? && !mounted.stdout.nil? && !mounted.stdout.empty?
     end
+
+    def md5sum
+      return perform_checksum(:md5) unless @backend.os.family == 'windows'
+
+      perform_checksum_windows(:md5)
+    end
+
+    def sha256sum
+      return perform_checksum(:sha256) unless @backend.os.family == 'windows'
+
+      perform_checksum_windows(:sha256)
+    end
+
+    private
+
+    def md5_command
+      case @backend.os.family
+      when 'darwin'
+        # `-r` reverses output to match `md5sum`
+        'md5 -r'
+      when 'solaris'
+        'digest -a md5'
+      else
+        'md5sum'
+      end
```

Comment:
I think we need to think about the case where those commands are not available. In those case I would like to fall back to the old methods

Reply:
Sounds good, added in last commit. Let me know what you think @chris-rock.